nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -174,6 +177,19 @@ ProtectedNiFiProperties readProtectedPropertiesFromDisk(File file) {
             rawProperties.load(inStream);
             logger.info("Loaded {} properties from {}", rawProperties.size(), file.getAbsolutePath());
 
+            // Trim whitespace from each property. If property is multi-line, remove anything after the first line break.
+            Set<String> keys = rawProperties.stringPropertyNames();
+            Iterator<String> itr = keys.iterator();
+            final Pattern whitespaceRegex = Pattern.compile("\\s+$");
+            while(itr.hasNext()){
```

Comment:
Is there a reason for using an `Iterator` as opposed to using a for loop over the `keys`?
```suggestion
            for (final String key : keys) {
```

Reply:
I think I'd been under the impression an Iterator was the preferred method of looping over a collection (my Java is rusty), but this is simpler.